clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -423,11 +433,12 @@ private boolean maybeSendTransactionalRequest() {
             if (!accumulator.flushInProgress())
                 accumulator.beginFlush();
         }
+    }
 
-        TransactionManager.TxnRequestHandler nextRequestHandler = transactionManager.nextRequestHandler(accumulator.hasIncomplete());
-        if (nextRequestHandler == null)
-            return false;
-
+    /**
+     * Enqueue a FindCoordinatorRequest if needed, otherwise send the next request
+     */
+    private boolean maybeSendTransactionalRequest(TransactionManager.TxnRequestHandler nextRequestHandler) {
         AbstractRequest.Builder<?> requestBuilder = nextRequestHandler.requestBuilder();
         while (!forceClose) {
             Node targetNode = null;
```

Comment:
Minor simplification we can do below:
```java
if (targetNode == null || !awaitReady(client, targetNode, time, requestTimeoutMs)) {
  transactionManager.lookupCoordinator(nextRequestHandler);
  break;                        
}
```

Reply:
A helper for `awaitReady` might be useful as well. Might be a chance to consolidate the `awaitLeastLoadedNodeReady` path.